### ISSUE TITLE
PlaceholderAPI support

### ISSUE THREAD

**Issue by turikhay** [Nuytemans-Dieter/BetterSleeping#77]
Hi!

I believe the philosophy of placeholders is that they are always computed relative to the player they refer to.
BetterSleeping is not following this philosophy: placeholders are always computed relative to the `<receiver>`, not the `<user>`.

I think you might need to see the [example from the PlaceholderAPI wiki](https://github.com/PlaceholderAPI/PlaceholderAPI/wiki/Hook-into-PlaceholderAPI#using-placeholders-from-placeholderapi-in-your-plugin): it is computed relative to the player that joined the game, not the player that going to receive the message.

**Comment by Nuytemans-Dieter** [comment]
Hi @turikhay, thank you for your input! I really appreciate it.

I assumed that the BetterSleeping placeholders would only be used for displaying the current status information to a player, as players in different worlds have different sleeping counts (sleeping, required and total needed). A player in a different world should be shown different information. That's why I went this way, but I'd like more insight as I have never used PlaceholderAPI.

Could you give an example of how you use BetterSleeping's placeholders?

Kind regards

**Comment by turikhay** [comment]
Yeah, of course. For example I use it to show the player's prefix and suffix:
```yaml
# This shows intended prefix and suffix:
good_morning: "Good morning! Have a great day, %vault_prefix%<receiver>%vault_suffix%&r&e!"

# This does not:
cancelled: "%vault_prefix%<user>%vault_suffix%&r&e left their bed!"
amount_left: "%vault_prefix%<user>%vault_suffix%&r&e is falling asleep."
```

**Comment by Nuytemans-Dieter** [comment]
Ooh, I don't know how I totally missed your point. That's would be a great improvement to the PlaceholderAPI support.

Unfortunately, I can only replace all PlaceholderAPI placeholders by providing a single user. Support for both `<receiver>` and `<user>` is not possible as far as I know. 

Changing the placeholders from being computed against `<user>` instead of `<receiver>` is a possibility (albeit quite hard with my current codebase) and would break BetterSleeping placeholders.

However, I will look into this and try to come up with a solution!

Thank you so much for this suggestion and clarification!!

**Comment by Nuytemans-Dieter** [comment]
Fixed in 57ae629add574f3a707687ae8e06ab10eab6ea1a
